packages/evm/contracts/BoostCore.sol

```diff
@@ -553,11 +553,15 @@ contract BoostCore is Ownable, ReentrancyGuard {
     }
 
     // Helper function to get the balance of the asset depending on its type
-    function _getAssetBalance(IncentiveDisbursalInfo storage incentive) internal view returns (uint256) {
+    function _getAssetBalance(
+        IncentiveDisbursalInfo storage incentive,
+        BoostLib.Boost storage boost,
+        uint256 incentiveId_
```

Comment:
might be better to pass in the incentive contract instead of boost and incentive id? or is this okay?

Reply:
I changed it to use incentive contract. Seems a lot cleaner.